practice-app/server/src/api/controllers/user/getAttendedEvents.js

```diff
@@ -0,0 +1,36 @@
+import mongoose from 'mongoose';
+import { Event } from '../../../models/index.js';
+import { User } from '../../../models/index.js';
+
+export default async (req, res) => {
+
+    const user_id = req.query.user_id;
+    if (!user_id || !mongoose.Types.ObjectId.isValid(req.query.user_id)) {
+        return res.status(400).json({"resultMessage": "Please provide a valid user id."});
+    }
+
+    const user = await User.find({ _id: user_id})
```

Comment:
You can use findOne method instead of find and so that you will not have to check first element of the array and directly control as "if(!user)".

Reply:
This is just an optimization that we can consider seriously in the next term while we are implementing our main project. For now, for the practical app, the implementation you did is quite enough and we are able to get the expected result. So, I'm resolving the conversation.